cmd/kubeadm/app/util/config/initconfiguration.go

```diff
@@ -380,6 +380,11 @@ func documentMapToInitConfiguration(gvkmap kubeadmapi.DocumentMap, allowDeprecat
 	// If ClusterConfiguration was given, populate it in the InitConfiguration struct
 	if clustercfg != nil {
 		initcfg.ClusterConfiguration = *clustercfg
+
+		// TODO: Workaround for missing v1beta3 ClusterConfiguration timeout conversion. Remove this conversion once the v1beta3 is removed
+		if clustercfg.APIServer.TimeoutForControlPlane.Duration != 0 && clustercfg.APIServer.TimeoutForControlPlane.Duration != kubeadmconstants.ControlPlaneComponentHealthCheckTimeout {
+			initcfg.Timeouts.ControlPlaneComponentHealthCheck.Duration = clustercfg.APIServer.TimeoutForControlPlane.Duration
+		}
```

Comment:
@neolit123 As you already pointed out, it is not possible to do the conversion with the `apimachinery conversion`. I did look at the call hierarchy and this is the best place I came up with to apply the fix. This change would obviously violate the principle that this conversion should be done in the conversion layer. I tested that with v1beta3 and v1beta4 with and without InitConfiguration and it seems to work fine. I'm not sure whether applying the change is worth it if v1beta3 is to removed in the next release. If not I would just close this PR and wait for the next release